recipes/opencv/4.x/conanfile.py

```diff
@@ -321,7 +321,7 @@ def add_components(components):
                             self.cpp_info.components[conan_component].libdirs.append("lib")
                             self.cpp_info.components[conan_component].libs += tools.collect_libs(self)
 
-                if self.settings.os == "iOS":
+                if self.settings.os == "iOS" or tools.os_info.is_macos:
```

Comment:
Hi, `tools.os_info` returns information about the _build_ machine, while `package_info` should return information about the _host_ libraries that the consumer is going to use/link. In a cross-building scenario we might be running on MacOS but compiling binaries for other OS, in that scenario we don't want this condition to evaluate to `True`. 

Hope this distinction is clear.

Reply:
Fixed.